src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -113,11 +134,331 @@ public void Initialize(ChainedHeader consensusTip)
                 throw new ConsensusException("INVALID_NETWORK");
             }
 
+            // Initialize local tip claim with consensus tip.
             this.AddOrReplacePeerTip(LocalPeerId, consensusTip.HashBlock);
+            
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <summary>Gets the consensus tip.</summary>
+        private ChainedHeader GetConsensusTip()
```

Comment:
Whats the reason not to use ChainState?

Reply:
1 dependency less

Reply:
Also it's a protection against chainstate.consensus tip being changed from some place we didnt want it to be changed